infra/modules/providers/azure/app-monitoring/main.tf

```diff
@@ -0,0 +1,36 @@
+resource "azurerm_monitor_action_group" "appmonitoring" {
+  count               = "${var.action_group_email_receiver == "" ? 0 : 1}"
+  name                = "${var.action_group_name}"
+  resource_group_name = "${var.resource_group_name}"
+  short_name          = "${var.action_group_short_name}"
+
+  email_receiver {
+    name          = "${var.action_group_email_receiver_name}"
+    email_address = "${var.action_group_email_receiver}"
+  }
+}
+
+resource "azurerm_monitor_metric_alert" "appmonitoring" {
+  count               = "${var.action_group_email_receiver  == "" ? 0 : 1}"
+  name                = "${var.metric_alert_name}"
+  resource_group_name = "${azurerm_monitor_action_group.appmonitoring.resource_group_name}"
+  scopes              = ["${var.resource_ids}"]
+
+  criteria {
+    metric_namespace = "${var.metric_alert_criteria_namespace}"
+    metric_name      = "${var.metric_alert_criteria_name}"
+    aggregation      = "${var.metric_alert_criteria_aggregation}"
+    operator         = "${var.metric_alert_criteria_operator}"
+    threshold        = "${var.metric_alert_criteria_threshold}"
+
+    dimension {
+      name     = "Instance"
+      operator = "Include"
+      values   = ["*"]
```

Comment:
[question for a curious reader] Not seeing much about what this means in the docs [here](https://www.terraform.io/docs/providers/azurerm/r/monitor_metric_alert.html). Do you have more context?

Reply:
These are options for scaling your alert metrics out to server instances available according to the chosen app-service plan subscription level. I prefixed the variables with "scaling" to provide more clarity.

Here's the link for more explanation.
https://docs.microsoft.com/en-us/azure/azure-monitor/platform/alerts-metric-overview#monitoring-at-scale-using-metric-alerts-in-azure-monitor